scipy/stats/tests/test_distributions.py

```diff
@@ -3236,6 +3236,48 @@ def test_logsf(self):
         y = stats.rayleigh.logsf(50)
         assert_allclose(y, -1250)
 
+    @pytest.mark.parametrize("floc", [-1, 0, 1, 2, 3])
+    @pytest.mark.parametrize("fscale", [.5, 1, 2, 3, 4])
+    def test_fit(self, floc, fscale):
+        data = stats.rayleigh.rvs(size=250, loc=floc, scale=fscale)
+
+        def scale_mle(data, floc):
+            return (np.sum((data - floc) ** 2) / (2 * len(data))) ** .5
+
+        scale_expect = scale_mle(data, floc)
+        # when `floc` is provided, direct MLE equation is used
+        loc, scale = stats.rayleigh.fit(data, floc=floc)
+        assert_equal(loc, floc)
+        assert_equal(scale, scale_expect)
+        # when `fscale` is fixed, standard numerical optimization is used.
+        loc, scale = stats.rayleigh.fit(data, fscale=.6)
+        assert_equal(scale, .6)
+
+        # with both parameters free, one dimentional optimization is done
+        # over a new function that takes into account the dependent relation
+        # of `scale` to `loc`.
+        loc, scale = stats.rayleigh.fit(data)
+        # test that `scale` is defined by its relation to `loc`
+        assert_equal(scale, scale_mle(data, loc))
+
+        # test that the objective function result of the analytical MLEs is
+        # less than or equal to that of the numerically optimized estimate
+        loc_scale_opt = super(type(stats.rayleigh), stats.rayleigh).fit(data)
+
+        # obtain objective function with same method as `rv_continuous.fit`
+        args = [data, (stats.rayleigh._fitstart(data), )]
+        func = stats.rayleigh._reduce_func(args, {})[1]
+
+        ll_mle = func((loc, scale), data)
+        ll_opt = func(loc_scale_opt, data)
```

Comment:
Please don't use "mle" or "opt" in variable names to distinguish between the superclass fit log-likelihood and overriden fit log-likelihood. Both the subclass and superclass fit methods perform `MLE`; both of them _opt_imize the log-likelihood.  They just differ in how they do it.